models/pod.go

```diff
@@ -143,8 +143,8 @@ func (pod *Pod) Parse(p *core_v1.Pod) {
 	pod.ServiceAccountName = p.Spec.ServiceAccountName
 }
 
-func isIstioProxy(pod *core_v1.Pod, container *core_v1.Container, conf *config.Config) bool {
-	if pod.Namespace != conf.IstioNamespace {
+func isIstioProxy(pod *core_v1.Pod, container *core_v1.Container, conf config.Config, isControlPlane func(cluster, namespace string) bool) bool {
+	if !isControlPlane("", pod.Namespace) {
```

Comment:
It looks like we're now passing `IsControlPlane` to a lot of places that previously had `conf.IstioNamespace` but I wonder if we need this check at all? This check in particular seems to say that if a pod is in the `istio-system` namespace then it's not a proxy which is not true.

Reply:
>  seems to say that if a pod is in the istio-system namespace then it's not a proxy

I think it's saying that if a pod IS NOT in a control plane namespace then it IS NOT an istio-proxy. In other words, it has to be in a control plane namespace.

I'm not 100% sure that this function is totally correct, but that test has been there for 4 years, so I don't think I want to change it as part of this PR.